node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/identity/AthenzCredentialsMaintainer.java

```diff
@@ -47,185 +52,165 @@
+            Instant now = clock.instant();
             Instant expiry = certificate.getNotAfter().toInstant();
             if (isCertificateExpired(expiry, now)) {
-                log.info(String.format("Certificate has expired (expiry=%s)", expiry.toString()));
-                registerIdentity();
+                context.log(logger, "Certificate has expired (expiry=%s)", expiry.toString());
+                registerIdentity(context, privateKeyFile, certificateFile, identityDocumentFile);
                 return;
             }
+
             Duration age = Duration.between(certificate.getNotBefore().toInstant(), now);
             if (shouldRefreshCredentials(age)) {
-                log.info(String.format("Certificate is ready to be refreshed (age=%s)", age.toString()));
-                if (shouldThrottleRefreshAttempts(now)) {
-                    log.warning(String.format("Skipping refresh attempt as last refresh was on %s (less than %s ago)",
-                                              lastRefreshAttempt.toString(), REFRESH_BACKOFF.toString()));
+                context.log(logger, "Certificate is ready to be refreshed (age=%s)", age.toString());
+                if (shouldThrottleRefreshAttempts(context, now)) {
+                    context.log(logger, LogLevel.WARNING, String.format(
+                            "Skipping refresh attempt as last refresh was on %s (less than %s ago)",
+                            lastRefreshAttempt.toString(), REFRESH_BACKOFF.toString()));
                     return;
                 } else {
-                    lastRefreshAttempt = now;
-                    refreshIdentity();
+                    lastRefreshAttempt.put(context.containerName(), now);
+                    refreshIdentity(context, privateKeyFile, certificateFile, identityDocumentFile);
                     return;
                 }
             }
-            log.debug("Certificate is still valid");
+            context.log(logger, LogLevel.DEBUG, "Certificate is still valid");
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
-    public void clearCredentials() {
-        if (!enabled) return;
-        try {
-            if (Files.deleteIfExists(privateKeyFile))
-                log.info(String.format("Deleted private key file (path=%s)", privateKeyFile));
-            if (Files.deleteIfExists(certificateFile))
-                log.info(String.format("Deleted certificate file (path=%s)", certificateFile));
-            if (Files.deleteIfExists(identityDocumentFile))
-                log.info(String.format("Deleted identity document file (path=%s)", certificateFile));
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
+    public void clearCredentials(NodeAgentContext context) {
+        FileFinder.files(context.pathOnHostFromPathInNode(CONTAINER_SIA_DIRECTORY))
```

Comment:
This works similarly to `find dir -type f -exec rm -r {} \;` (will delete all files under container's `/var/lib/sia`